Documentation/network/servicemesh/ingress-to-gateway/tls-migration.rst

```diff
@@ -0,0 +1,161 @@
+
+            apiVersion: networking.k8s.io/v1
+            kind: Ingress
+            metadata:
+            name: tls-ingress
+            namespace: default
+            spec:
+              tls:
+              - hosts:
+                - bookinfo.cilium.rocks
+                - hipstershop.cilium.rocks
+                secretName: demo-cert
+
+    .. group-tab:: Gateway API
+
+        In the Gateway API, TLS termination is a property of the Gateway listener, and similarly to the Ingress, a TLS certificate and key are also stored in a Secret.
+
+        .. code-block:: shell-session
+
+            apiVersion: gateway.networking.k8s.io/v1beta1
+            kind: Gateway
+            metadata:
+              name: tls-gateway
+            spec:
+              gatewayClassName: cilium
+            listeners:
+            - name: bookinfo.cilium.rocks
+              protocol: HTTPS
+              port: 443
+              hostname: "bookinfo.cilium.rocks"
+              tls:
+                certificateRefs:
+                - kind: Secret
+                    name: demo-cert
+            - name: hipstershop.cilium.rocks
+              protocol: HTTPS
+              port: 443
+              hostname: "hipstershop.cilium.rocks"
+              tls:
+                certificateRefs:
+                - kind: Secret
+                    name: demo-cert
+
+- Host-header-based Routing Rules
+
+.. tabs::
+
+    .. group-tab:: Ingress
+        
+        The Ingress API uses the term *host*.
+        With Ingress, each host has separate routing rules.
+
+        .. code-block:: shell-session
+
+            apiVersion: networking.k8s.io/v1
+            kind: Ingress
+            metadata:
+            name: tls-ingress
+            namespace: default
+            spec:
+            ingressClassName: cilium
+            rules:
+            - host: hipstershop.cilium.rocks
+                http:
+                paths:
+                - backend:
+                    service:
+                        name: productcatalogservice
+                        port:
+                        number: 3550
+                    path: /hipstershop.ProductCatalogService
+                    pathType: Prefix
+
+    .. group-tab:: Gateway API
+
+        The Gateway API uses the *hostname* term.
+        The host-header-based routing rules map to the hostnames of the HTTPRoute. 
+        In the HTTPRoute, the routing rules apply to all hostnames.
+
+        The hostnames of an HTTPRoute must match the hostname of the Gateway listener. Otherwise, the listener will ignore the routing rules for the unmatched hostnames.
+
+        .. code-block:: shell-session
+
+            ---
+            apiVersion: gateway.networking.k8s.io/v1beta1
+            kind: HTTPRoute
+            metadata:
+            name: hipstershop-cilium-rocks
+            namespace: default
+            spec:
+            hostnames:
+            - hipstershop.cilium.rocks
```

Comment:
Note that setting the hostname like this will work in the way you expect, but maybe not for the reason you expect. :)

When you attach a HTTPRoute to a Gateway without using the `sectionName` field, then it will attempt to attach to _all_ Listeners in the Gateway. In this case, attaching to the Listener for `bookinfo.cilium.rocks` will fail because the HTTPRoute also specifies the hostname `hipstershop.cilium.rocks`, so the HTTPRoute will only attach to the `hipstershop.cilium.rocks` Listener.

However, you could also achieve the same effect by leaving out the `hostname` field from the HTTPRoute and setting the `sectionName` field in the `parentRef` to `hipstershop.cilium.rocks` (the `name` field of the Listener you want to attach to).

The example is fine as is, just thought you would appreciate the extra context.

Reply:
Yes, I didn't know you could `sectionName` like this - I will have to test this (all the scripts we use in the labs are like the one in the docs above).

Reply:
I'll raise this in the upstream meeting, see what other people are generally using.